src/Datadog.Trace/Logging/LibLogScopeEventSubscriber.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+
+namespace Datadog.Trace.Logging
+{
+    internal class LibLogScopeEventSubscriber : IDisposable
+    {
+        // Keep track of the active Scope (and its corresponding logging contexts)
+        // so if another Scope is deactivated (not the active Scope), no
+        // changes are made
+        private IScopeManager _scopeManager;
+        private Scope _activeScope;
+        private IDisposable _activeTraceContext;
+        private IDisposable _activeSpanContext;
+
+        public LibLogScopeEventSubscriber(IScopeManager scopeManager)
+        {
+            _scopeManager = scopeManager;
+            _scopeManager.ScopeActivated += OnScopeActivated;
+            _scopeManager.ScopeDeactivated += OnScopeDeactivated;
+
+            _activeTraceContext = LogProvider.OpenMappedContext(CorrelationIdentifier.TraceIdKey, 0, destructure: false);
+            _activeSpanContext = LogProvider.OpenMappedContext(CorrelationIdentifier.SpanIdKey, 0, destructure: false);
+        }
+
+        public void OnScopeActivated(object sender, ScopeEventArgs scopeEventArgs)
+        {
+            // Contexts MUST be closed in reverse order of opening
+            _activeSpanContext?.Dispose();
+            _activeTraceContext?.Dispose();
```

Comment:
Is the trace context always fully in step with the span context? Or should the trace context match up with only the root span?

Reply:
These two objects are always in sync. The intent is that we are querying the active span (if any) and extracting both its SpanId and TraceId, and these "Context" objects are the logging context setters for each of the properties. Does that make sense? And would better naming make this intent more clear?

Reply:
It's kind of odd to read in my brain, because the `TraceId` will remain the same through the entire lifetime of a root span,
whereas a `SpanId` changes with each scope change.

Reply:
I see what you're saying and yeah that is the standard behavior. I think it is possible, though extremely unlikely, for a new span to be activated with a different context using the `Tracer.StartActive` method so I think the safest thing to do is to check and update the TraceId every single time. If I'm incorrect and we don't need this flexibility, I'm happy to keep the trace context around and not dispose it every time.

Reply:
If it's always in step with the span context, why have two?

Reply:
When I get in the office today, I'll make a small change to check if the TraceId is the same before disposing it 👍

Reply:
After understanding LogProvider a bit more, I understand. This is good with me.

Reply:
I refactored the logging context code to make it a little more clear that the contexts here are setters for the key-value pairs within the logging context, but we still set/unset the pair every time we receive a new span. Let me know if there is anything more that you think should be done here.